### PR TITLE
Made paasta status and paasta metastatus return nonzero on ssh error

### LINKED ISSUE #718: paasta status exits 0 when it cannot ssh (should be nonzero)
```
$ env -i paasta status -c norcal-prod -s www_pages
Pipeline: https://jenkins.yelpcorp.com/view/services-www_pages

cluster: norcal-prod
    ERROR: could not find connectable master in cluster norcal-prod
    Output: ERROR cannot run check command ssh -A -n 10.44.6.103 sudo paasta_serviceinit -h
    Return code was 255 which is an unknown failure.
    HINT: Talk to #operations and pastebin this output
    Output from check command: Permission denied (publickey).
    ssh_exchange_identification: Connection closed by remote host


$ echo $?
0
```

### PR DESCRIPTION
**Pull request by mjksmith**
Fixes #718 

Also upgraded `mock` to `2.0.0` to fix a bug with `side_effect` that caused some unexpected behaviour.

### CODE REVIEW

File: paasta_tools/cli/cmds/metastatus.py [hunk 1]
```diff
@@ -121,15 +125,19 @@ def paasta_metastatus(args):
...
+                    system_paasta_config=system_paasta_config,
+                    humanize=args.humanize,
+                    groupings=args.groupings,
+                    verbose=args.verbose,
+                )
             )
         else:
             print "Cluster %s doesn't look like a valid cluster?" % args.clusters
             print "Try using tab completion to help complete the cluster name"
+    return all([return_code == 0 for return_code in return_codes])
```
Comment (mattmb) [inline paasta_tools/cli/cmds/metastatus.py]: Is this doing what you expect? Won't it return True if everything returns 0 and then end up with `sys.exit(True)`?

**Comment by mjksmith** [comment]
@mattmb the reason your fix wasn't caught by testing is because I didn't prefix my tests with 'test_' -_-

Always make sure your tests can fail, kids.

File: paasta_tools/cli/cmds/metastatus.py [hunk 2]
```diff
@@ -121,15 +125,19 @@ def paasta_metastatus(args):
...
+                    system_paasta_config=system_paasta_config,
+                    humanize=args.humanize,
+                    groupings=args.groupings,
+                    verbose=args.verbose,
+                )
             )
         else:
             print "Cluster %s doesn't look like a valid cluster?" % args.clusters
             print "Try using tab completion to help complete the cluster name"
+    return 0 if all([return_code == 0 for return_code in return_codes]) else 1
```
Comment (solarkennedy) [inline paasta_tools/cli/cmds/metastatus.py]: How about max() ?

**Comment by solarkennedy** [comment]
ship

**Pull request merged**